src/game/g_weapon.c

```diff
@@ -1253,35 +1253,35 @@ void G_UpdateZaps( int msec )
 
           //let's not have the damage numbers inflated by a high msec value, thank you very much
           if (trueDamage + zap->damageUsed > LEVEL2_AREAZAP_DMG )
-          {
             trueDamage = LEVEL2_AREAZAP_DMG - zap->damageUsed;
-          }
+            
           zap->damageUsed += trueDamage;
         }
 
-        for( j = 0; j < zap->numTargets; j++ )
+        for( j = 0; j < zap->numTargets ; j++ )
         {
           gentity_t *source;
           gentity_t *target = zap->targets[ j ];
           int       damage;
-          float     damageFalloffFac;
-          float     armourTotal = 0;
+          float     armourFac = 0;
           vec3_t    forward;
 
           if( j == 0 )
             source = zap->creator;
           else
             source = zap->targets[ j - 1 ];
 
-          if( BG_InventoryContainsUpgrade( UP_BATTLESUIT, target->client->ps.stats ) ) armourTotal = BSUIT_NL_AVERAGE;
-          else
-          {
-            if( BG_InventoryContainsUpgrade( UP_HELMET, target->client->ps.stats ) ) armourTotal += HELMET_NL_AVERAGE;
-            if( BG_InventoryContainsUpgrade( UP_LIGHTARMOUR, target->client->ps.stats ) ) armourTotal += LIGHTARMOUR_NL_AVERAGE;
+          if(target->s.eType == ET_PLAYER ){
```

Comment:
Change the line to `if( target->client )` for consistency with the rest of zap code. You generally don't check for players by looking at the entity type, but by looking for and inside the client structure.